crates/ruff/src/rules/pycodestyle/rules/logical_lines/missing_whitespace_around_operator.rs

```diff
@@ -86,13 +86,19 @@ pub(crate) fn missing_whitespace_around_operator(
             );
 
             NeedsSpace::from(!slash_in_func)
-        } else if kind.is_unary() || kind == TokenKind::DoubleStar {
+        } else if kind.is_unary()
+            || matches!(
+                kind,
+                TokenKind::Star | TokenKind::DoubleStar | TokenKind::At
+            )
+        {
             let is_binary = prev_token.map_or(false, |prev_token| {
                 let prev_kind = prev_token.kind();
```

Comment:
An you explain the fix. It's unclear to me how it works. I would have expected that the @ always goes into the else branch that requires no space, and thus, doesn't create any diagnostics

Reply:
`@` can be used as a binary operator in Python -- it represents matrix multiplication:

```
#: E225
i = 1 @2
#: E225
i = 1@ 2
```

So equivalently to `*` or `+`, we need to detect whether we're seeing a decorator, or a binary operator.

Reply:
Oh wow, I didn't know! 

Hmm, I'm wondering right now if my refactored implementation misses the handling for `WS_OPTIONAL`

https://github.com/PyCQA/pycodestyle/blob/f6500e2170406b664a721b5e3a0f5d66bcb757e1/pycodestyle.py#L108

Reply:
Hmm, I will take a look in a quick follow-up PR (or you can, if you're already working on it?).

Reply:
I'm not and would appreciate it if you can have a look. I believe we're missing the whole optional branch because `pycodestyle` has no explicit handling of the `@` in the `unary` branch.

Reply:
Okay, I now remember briefly what is happening. The arithmetic and bitwise aren't optional. It's just that they are different rules (handled here https://github.com/PyCQA/pycodestyle/blob/f6500e2170406b664a721b5e3a0f5d66bcb757e1/pycodestyle.py#LL914C26-L920C66)

What I don't understand right now is why `@` would require any special treatment (more specifically, why we need where pycodestyle has none)